catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelper.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard.util;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.util.impl.SortedServiceReferenceList;
+
+/**
+ * Support class for working with {@link Attribute}s and {@link Metacard}s.
+ */
+public class AttributeHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeHelper.class);
+
+    private final Map<String, AttributeDescriptor> systemDescriptors;
+
+    /**
+     * Default constructor for use when the basic attributes are sufficient.
+     */
+    public AttributeHelper() {
+        this.systemDescriptors = new HashMap<>();
+    }
+
+    /**
+     * Constructor given service references to available {@link MetacardType}s.
+     *
+     * @param serviceReferences A list of {@link ServiceReference}s representing exposed
+     *                          {@link MetacardType}s.
+     */
+    public AttributeHelper(SortedServiceReferenceList serviceReferences) {
+        this.systemDescriptors = new HashMap<>();
+
+        BundleContext bundleContext = FrameworkUtil.getBundle(AttributeHelper.class)
+                .getBundleContext();
```

Comment:
so i forget which.. but either `getBundle` or `getBundleContext` (maybe both) can return null. (i find I encounter it in unit tests) Which will cause a NPE below